xen/drivers/passthrough/iommu.c

```diff
@@ -249,22 +249,39 @@ void iommu_domain_destroy(struct domain *d)
     arch_iommu_domain_destroy(d);
 }
 
-int iommu_map_page(struct domain *d, unsigned long gfn, unsigned long mfn,
-                   unsigned int flags)
+int iommu_map_pages(struct domain *d, unsigned long gfn, unsigned long mfn,
+                    unsigned long page_count, unsigned int flags)
```

Comment:
I don't think so, size_t is intended to use for size, for example size_t page_size = 4096, but unsigned long page_count = 1.